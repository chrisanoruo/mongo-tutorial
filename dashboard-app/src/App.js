import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Row, Col, Card, Layout, Spin, Statistic, Table } from "antd";
import "antd/dist/antd.css";
import "./index.css";
import cubejs from "@cubejs-client/core";
import { QueryRenderer } from "@cubejs-client/react";
import { Chart, Axis, Tooltip, Geom, Coord, Legend } from "bizcharts";
import moment from "moment";

const AppLayout = ({ children }) => (
  <Layout>
    <Layout.Header>
      <div
        style={{
          float: "left"
        }}
      >
        <h2
          style={{
            color: "#fff",
            margin: 0,
            marginRight: "1em"
          }}
        >
          My Dashboard
        </h2>
      </div>
    </Layout.Header>
    <Layout.Content
      style={{
        padding: "0 25px 25px 25px",
        margin: "25px"
      }}
    >
      {children}
    </Layout.Content>
  </Layout>
);

const Dashboard = ({ children }) => (
  <Row type="flex" justify="space-around" align="top" gutter={24}>
    {children}
  </Row>
);

const DashboardItem = ({ children, title }) => (
  <Col span={24} lg={12}>
    <Card
      title={title}
      style={{
        marginBottom: "24px"
      }}
    >
      {children}
    </Card>
  </Col>
);

const stackedChartData = resultSet => {
  const data = resultSet
    .pivot()
    .map(({ xValues, yValuesArray }) =>
      yValuesArray.map(([yValues, m]) => ({
        x: resultSet.axisValuesString(xValues, ", "),
        color: resultSet.axisValuesString(yValues, ", "),
        measure: m && Number.parseFloat(m)
      }))
    )
    .reduce((a, b) => a.concat(b));
  return data;
};

const pieRender = ({ resultSet }) => (
  <Chart height={400} data={resultSet.chartPivot()} forceFit>
    <Coord type="theta" radius={0.75} />
    {resultSet.seriesNames().map(s => (
      <Axis name={s.key} />
    ))}
    <Legend position="right" />
    <Tooltip />
    {resultSet.seriesNames().map(s => (
      <Geom type="intervalStack" position={s.key} color="category" />
    ))}
  </Chart>
);

const API_URL = "http://localhost:4000";
const cubejsApi = cubejs(
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjU1NzY0NjMsImV4cCI6MTU2NTY2Mjg2M30.OIF1sF9ROBCl_PNdxQrggOl9hCnCGY7QN7MVf-khSWs",
  {
    apiUrl: API_URL + "/cubejs-api/v1"
  }
);

const renderChart = Component => ({ resultSet, error }) =>
  (resultSet && <Component resultSet={resultSet} />) ||
  (error && error.toString()) || <Spin />;

function App() {
  return (
    <div className="App">
      <AppLayout>
        <Dashboard>
          <DashboardItem>
            <QueryRenderer
              query={{
                measures: ["Zips.count"],
                timeDimensions: [],
                dimensions: ["Zips.state"],
                filters: []
              }}
              cubejsApi={cubejsApi}
              render={renderChart(pieRender)}
            />
          </DashboardItem>
        </Dashboard>
      </AppLayout>
    </div>
  );
}

export default App;
