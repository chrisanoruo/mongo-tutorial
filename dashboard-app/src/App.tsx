import React from 'react';
import logo from './logo.svg';
import 'antd/dist/antd.css';
import { Row, Col, Card, Layout, Spin } from 'antd';

import cubejs from '@cubejs-client/core';
import { QueryRenderer } from '@cubejs-client/react';
import { Chart, Axis, Tooltip, Geom, Coord, Legend } from 'bizcharts';

import './App.css';

const renderChart = (resultSet: any) => (
  <Chart height={400} data={resultSet.chartPivot()} forceFit>
    <Coord type="theta" radius={0.75} />
    <Axis name="Zips.count" />
    <Legend position="bottom" name="category" />
    <Tooltip showTitle={false} />
    <Geom type="intervalStack" position="Zips.count" color="x" />
  </Chart>
);

const cubejsApi = cubejs(
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NDkzMDk3NzMsImV4cCI6MTU0OTM5NjE3M30.eXEdfUa_ek2V9MlGTpBMOd_AFfs8laaZj8ZsuM1wqqo",
  { apiUrl: "http://localhost:4000/cubejs-api/v1" }
);

const { Header, Footer, Sider, Content } = Layout;

const App: React.FC = () => {
  
  return (
    <Layout>
      <Header>
        <h2 style={{ color: "#fff" }}>MongoDB Dashboard</h2>
      </Header>
      <Content style={{ padding: "25px", margin: "25px" }}>
        <Row type="flex" justify="space-around" align="top" gutter={24}>
          <Col span={24} lg={12}>
            <Card title="Zip count by state" style={{ marginBottom: "24px" }}>
              <QueryRenderer
                query={{ measures: ["Zips.count"], dimensions: ["Zips.state"] }}
                cubejsApi={cubejsApi}
                render={({ resultSet, error }: any) =>
                (resultSet && renderChart(resultSet)) ||
                  (error && error.toString()) || <Spin />
                }
                />
            </Card>
          </Col>
        </Row>
      </Content>
    </Layout>
  );
}

export default App;
