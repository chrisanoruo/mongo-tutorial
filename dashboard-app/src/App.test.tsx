import React from 'react';
import ReactDOM from 'react-dom';
// import { shallow } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

configure({ adapter: new Adapter() });

// const App = () => {};

// it('should be ok', () => {
//   const flag = 1;
//   expect(flag).toBe(1);
// })

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});