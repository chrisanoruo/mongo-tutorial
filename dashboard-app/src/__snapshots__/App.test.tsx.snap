// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Adapter>
  <Adapter>
    <h2
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      MongoDB Dashboard
    </h2>
  </Adapter>
  <Adapter
    style={
      Object {
        "margin": "25px",
        "padding": "25px",
      }
    }
  >
    <Row
      align="top"
      gutter={24}
      justify="space-around"
      type="flex"
    >
      <Col
        lg={12}
        span={24}
      >
        <Card
          style={
            Object {
              "marginBottom": "24px",
            }
          }
          title="Zip count by state"
        >
          <QueryRenderer
            cubejsApi={
              CubejsApi {
                "apiToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NDkzMDk3NzMsImV4cCI6MTU0OTM5NjE3M30.eXEdfUa_ek2V9MlGTpBMOd_AFfs8laaZj8ZsuM1wqqo",
                "apiUrl": "http://localhost:4000/cubejs-api/v1",
              }
            }
            query={
              Object {
                "dimensions": Array [
                  "Zips.state",
                ],
                "measures": Array [
                  "Zips.count",
                ],
              }
            }
            render={[Function]}
          />
        </Card>
      </Col>
    </Row>
  </Adapter>
</Adapter>
`;

exports[`test should pass 1`] = `
<Adapter>
  <Adapter>
    <h2
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      MongoDB Dashboard
    </h2>
  </Adapter>
  <Adapter
    style={
      Object {
        "margin": "25px",
        "padding": "25px",
      }
    }
  >
    <Row
      align="top"
      gutter={24}
      justify="space-around"
      type="flex"
    >
      <Col
        lg={12}
        span={24}
      >
        <Card
          style={
            Object {
              "marginBottom": "24px",
            }
          }
          title="Zip count by state"
        >
          <QueryRenderer
            cubejsApi={
              CubejsApi {
                "apiToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NDkzMDk3NzMsImV4cCI6MTU0OTM5NjE3M30.eXEdfUa_ek2V9MlGTpBMOd_AFfs8laaZj8ZsuM1wqqo",
                "apiUrl": "http://localhost:4000/cubejs-api/v1",
              }
            }
            query={
              Object {
                "dimensions": Array [
                  "Zips.state",
                ],
                "measures": Array [
                  "Zips.count",
                ],
              }
            }
            render={[Function]}
          />
        </Card>
      </Col>
    </Row>
  </Adapter>
</Adapter>
`;
